trio/_core/_run.py

```diff
@@ -169,7 +202,8 @@ def collapse_exception_group(excgroup):
         )
         return exceptions[0]
     elif modified:
-        return excgroup.derive(exceptions)
+        # derive() returns Any for some reason.
+        return excgroup.derive(exceptions)  # type: ignore[no-any-return]
```

Comment:
The signature in [argonholm/exceptiongroup](https://github.com/agronholm/exceptiongroup/blob/0878b8331a905f18b4d39b24b105b3e514ada65b/src/exceptiongroup/_exceptions.py#L213) is
```python
    def derive(self: Self, __excs: Sequence[_BaseExceptionT_co]) -> Self:
```
and in [typeshed](https://github.com/python/typeshed/blob/5e2fe131faae347e109b48f451e4b8521378d764/stdlib/builtins.pyi#L2109-L2113) it is
```python
# In reality it is `NonEmptySequence`:
@overload
def derive(self, __excs: Sequence[_ExceptionT]) -> ExceptionGroup[_ExceptionT]: ...
@overload
def derive(self, __excs: Sequence[_BaseExceptionT]) -> BaseExceptionGroup[_BaseExceptionT]: ...

```


I found https://github.com/agronholm/exceptiongroup/pull/40, https://github.com/python/typeshed/issues/9219, and https://github.com/python/typeshed/issues/9230 and reading those it seems that this should've been resolved?

It might be that `disallow_any_generics` would flag something further up if you enabled it

Reply:
Seems like it didn't. @Zac-HD as the resident expert on ExceptionGroups, would you mind weighing in what's going on here? Not a blocker for merging the PR though

Reply:
Looks like the `exceptiongroup` package hasn't done the equivalent of https://github.com/python/typeshed/issues/9219, which is my bad for not following up on https://github.com/agronholm/exceptiongroup/pull/40#discussion_r1024633541.

I think it's fine to ignore this for now, but if you wanted to open an upstream issue and revisit later that'd be great.

Reply:
Made an upstream issue! See https://github.com/agronholm/exceptiongroup/issues/76